server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -557,6 +573,82 @@ public String toString() {
         }
     }
 
+    /**
+     * Extension of the Clusters class for CCS uses. It adds the number of remote clusters
+     * included in a search and whether ccs_minimize_roundtrips is true.
+     *
+     * Like Clusters, this is an immutable class.
+     *
+     * Unlike Clusters, (for now at least) the new fields are not Writeable. They are only
+     * used during CCS searches for accounting on the primary coordinator.
+     */
+    public static class CcsClusters extends Clusters {
```

Comment:
The point of having the Clusters class in the first place is to handle CCS, so to have one extension of Clusters that is called CCSClusters is kinda weird :)  Looking at its usage, this is a very specific extension that is only used when minimized roundtrips is on. How about `MinimizedRoundtripsAwareClusters` or something like that?